gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/bottomsheet/GravatarQuickEditorBottomSheet.kt

```diff
@@ -100,41 +107,96 @@ internal fun GravatarQuickEditorBottomSheet(
+        if (modalBottomSheetState.currentDetent == Hidden) {
+            onDismiss(GravatarQuickEditorDismissReason.Finished)
+        }
+    }
+
     GravatarTheme {
         ModalBottomSheet(
-            onDismissRequest = { onDismiss(GravatarQuickEditorDismissReason.Finished) },
-            sheetState = modalBottomSheetState,
-            dragHandle = { QEDragHandle() },
-            containerColor = MaterialTheme.colorScheme.surface,
-            tonalElevation = 1.dp,
-            contentWindowInsets = { WindowInsets(0, 0, 0, 0) },
+            state = modalBottomSheetState,
         ) {
-            Surface(
-                modifier = Modifier.navigationBarsPadding(),
+            Scrim(
+                modifier = Modifier.clickable { modalBottomSheetState.currentDetent = Hidden },
+                scrimColor = MaterialTheme.colorScheme.scrim.copy(alpha = 0.32f),
+            )
+            Sheet(
+                modifier = Modifier
+                    .clip(RoundedCornerShape(topStart = 28.dp, topEnd = 28.dp))
+                    .background(MaterialTheme.colorScheme.surfaceColorAtElevation(1.dp))
+                    .widthIn(max = 640.dp)
+                    .fillMaxWidth()
+                    .padding(
+                        WindowInsets.navigationBars
+                            .only(WindowInsetsSides.Vertical)
+                            .asPaddingValues(),
+                    ),
             ) {
-                Column {
-                    QETopBar(
-                        onDoneClick = {
-                            coroutineScope.launch {
-                                modalBottomSheetState.hide()
-                                onDismiss(GravatarQuickEditorDismissReason.Finished)
-                            }
-                        },
-                    )
-                    content()
+                Surface(
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    tonalElevation = 1.dp,
+                ) {
+                    Column(
+                        horizontalAlignment = Alignment.CenterHorizontally,
+                    ) {
+                        QEDragHandle()
+                        QETopBar(
+                            onDoneClick = {
+                                coroutineScope.launch {
+                                    modalBottomSheetState.currentDetent = Hidden
+                                }
+                            },
+                        )
+                        content()
+                    }
                 }
             }
         }
     }
 }
 
+@Composable
+internal fun rememberGravatarModalBottomSheetState(
+    avatarPickerContentLayout: AvatarPickerContentLayout,
+): ModalBottomSheetState {
+    val windowHeightSizeClass = currentWindowAdaptiveInfo().windowSizeClass.windowHeightSizeClass
+    val peek = SheetDetent(identifier = "peek") { containerHeight, _ ->
```

Comment:
Ktlint complains when I use `Peek`, can we make an exception just for this?

Reply:
Here no, because I renamed it to lowercase `peek`. It complains when I use `Peek`.

Reply:
Are you talking about -> 

```kotlin
val Peek = SheetDetent(identifier = "peek") { containerHeight, _ ->
``` 

Why do you want to use uppercase there?

Reply:
To match the naming convention of the Detents defined by the library. Not a big deal though.

Reply:
Oh, I see. Yeah, I don't think it's a big deal.